frame/wormhole/issuing/s2s/src/tests.rs

```diff
@@ -18,126 +18,167 @@
+	let t = LegacyUnsignedTransaction {
+		nonce: U256::from(nonce),
+		gas_price: U256::from(1),
+		gas_limit: U256::from(gas_limit_call),
+		action: ethereum::TransactionAction::Call(contract),
+		value: U256::from(value),
+		input,
+	}
+	.sign(&alice.private_key);
+	assert_ok!(Ethereum::execute(alice.address, &t.into(), None,));
+}
+
+fn configure_mapping_token_factory(alice: &AccountInfo) {
+	let mapping_token_factory_address: H160 =
+		array_bytes::hex_into_unchecked("32dcab0ef3fb2de2fce1d2e0799d36239671f04a");
+	// initialize, then the owner is system account
+	let initialize: Vec<u8> = hex2bytes_unchecked("0x8129fc1c").to_vec();
+	alice_call(&alice, initialize, 2, mapping_token_factory_address, 0);
+	// setTokenContractLogic
+	let set_token_contract_logic0 = hex2bytes_unchecked("0x3c547e160000000000000000000000000000000000000000000000000000000000000000000000000000000000000000248e85939e48ca12a20cdf80e60d9e3d380ca7f9");
+	let set_token_contract_logic1 = hex2bytes_unchecked("0x3c547e160000000000000000000000000000000000000000000000000000000000000001000000000000000000000000248e85939e48ca12a20cdf80e60d9e3d380ca7f9");
+	alice_call(
+		&alice,
+		set_token_contract_logic0,
+		3,
+		mapping_token_factory_address,
+		0,
+	);
+	alice_call(
+		&alice,
+		set_token_contract_logic1,
+		4,
+		mapping_token_factory_address,
+		0,
+	);
 }
 
 #[test]
 fn register_and_issue_from_remote_success() {
 	let (pairs, mut ext) = new_test_ext(1);
 	let alice = &pairs[0];
 	ext.execute_with(|| {
-		let t = LegacyUnsignedTransaction {
-			nonce: U256::zero(),
-			gas_price: U256::from(1),
-			gas_limit: U256::from(0x100000),
-			action: ethereum::TransactionAction::Create,
-			value: U256::zero(),
-			input: hex2bytes_unchecked(TEST_CONTRACT_BYTECODE),
-		}
-		.sign(&alice.private_key);
-		assert_ok!(Ethereum::execute(alice.address, &t.into(), None,));
-		let mapping_token_factory_address: H160 =
-			array_bytes::hex_into_unchecked("32dcab0ef3fb2de2fce1d2e0799d36239671f04a");
-		assert_ok!(S2sIssuing::set_mapping_factory_address(
-			Origin::root(),
-			mapping_token_factory_address,
-		));
-		let remote_root_address = hex2bytes_unchecked(
-			"0xaaa5b780fa60c639ad17212d92e8e6257cb468baa88e1f826e6fe8ae6b7b700c",
-		);
-		let remote_backing_account: AccountId32 =
-			AccountId32::decode(&mut &remote_root_address[..]).unwrap_or_default();
-		let original_token_address = hex_into_unchecked("0000000000000000000000000000000000000002");
-		let token = TokenMetadata::new(
-			NATIVE_TOKEN_TYPE,
-			original_token_address,
-			[10u8; 32].to_vec(),
-			[20u8; 32].to_vec(),
-			18u8,
-		);
-		let drived_remote_backing_account: AccountId32 =
-			hex_into_unchecked("77c1308128b230173f735cb97d6c62e5d8eeb86b148ff8461835c836945b1d84");
-		let backing_address = <Test as s2s_issuing::Config>::ToEthAddressT::into_ethereum_id(
-			&drived_remote_backing_account,
-		);
+        alice_create(&alice, FromHex::from_hex(MAPPING_TOKEN_FACTORY_CONTRACT_BYTECODE).unwrap(), 0);
```

Comment:
Good Job. Does this test function only test the whole workflow of the s2s-issuing pallet? This pallet logic looks a bit complicated in my view, do we need to add more corner or exception tests? It's not too convincing for me to test such an important pallet with only one test.

Reply:
In the Issuing module, the main logic is inside the contract. And there is already a more comprehensive test inside the contract, so here we'd better not involve too much in the branching scenario inside the contract. https://github.com/darwinia-network/darwinia-bridges-sol/blob/master/contracts/wormhole/ test/0_test_s2s.js

Reply:
That makes sense.